apps/web/src/components/Shared/UserPreview.tsx

```diff
@@ -31,102 +33,130 @@ const UserPreview: FC<Props> = ({
+                <div className="w-10 h-8 rounded-lg shimmer" />
+              ) : following ? null : profileData?.followModule?.__typename === 'FeeFollowModuleSettings' ? (
+                <SuperFollow profile={profileData} setFollowing={setFollowing} />
+              ) : (
+                <Follow profile={profileData} setFollowing={setFollowing} />
+              ))}
+          </div>
+        )}
       </div>
       <div className="p-1 space-y-3">
         <UserName />
-        <div>
-          {profile?.bio && (
-            <div className={clsx(isBig ? 'text-base' : 'text-sm', 'mt-2', 'linkify break-words leading-6')}>
-              <Markup>{profile?.bio}</Markup>
+        {!loading ? (
+          <>
+            <div>
+              {profileData?.bio && (
+                <div
+                  className={clsx(isBig ? 'text-base' : 'text-sm', 'mt-2', 'linkify break-words leading-6')}
+                >
+                  <Markup>{profileData?.bio}</Markup>
+                </div>
+              )}
             </div>
-          )}
-        </div>
-        <div className="flex space-x-3 items-center">
-          <div className="flex items-center space-x-1">
-            <div className="text-base">{nFormatter(profile?.stats?.totalFollowing)}</div>
-            <div className="text-gray-500 text-sm">Following</div>
-          </div>
-          <div className="flex items-center space-x-1 text-md">
-            <div className="text-base">{nFormatter(profile?.stats?.totalFollowers)}</div>
-            <div className="text-gray-500 text-sm">Followers</div>
-          </div>
-        </div>
-        {currentProfile && <MutualFollowers profile={profile} variant="xs" />}
+            <div className="flex space-x-3 items-center">
+              <div className="flex items-center space-x-1">
+                <div className="text-base">{nFormatter(profileData?.stats?.totalFollowing)}</div>
+                <div className="text-gray-500 text-sm">Following</div>
+              </div>
+              <div className="flex items-center space-x-1 text-md">
+                <div className="text-base">{nFormatter(profileData?.stats?.totalFollowers)}</div>
+                <div className="text-gray-500 text-sm">Followers</div>
+              </div>
+            </div>
+            {currentProfile && <MutualFollowers profile={profileData} variant="xs" />}
+          </>
+        ) : (
+          <Loader message="Loading" />
+        )}
       </div>
     </>
   );
 
   const onPreviewEnd = () => {
-    setShowPreview(false);
+    // setShowPreview(false);
   };
 
-  const onPreviewStart = () => {
+  const onPreviewStart = async () => {
     setShowPreview(true);
+    if (profileData.id == '') {
+      setLoading(true);
+      const { data } = await loadProfile({
+        variables: { request: { handle: formatHandle(profileData?.handle, true) } }
+      });
+      setLoading(false);
+      const lazyProfile = data?.profile;
```

Comment:
It would be great if we just keep `profile`, `lazyProfile` sounds weird 🤣

Reply:
lazyprofile can be state name, as we are getting data using lazy query